core/node/eigenda_proxy/src/eigenda_client.rs

```diff
@@ -0,0 +1,271 @@
+use std::{
+    sync::Arc,
+    time::{Duration, Instant},
+};
+
+use rlp::decode;
+use tokio::{sync::Mutex, time::interval};
+use tonic::transport::{Channel, ClientTlsConfig};
+use zksync_config::configs::da_client::eigen_da::DisperserConfig;
+
+use crate::{
+    blob_info::BlobInfo,
+    disperser::{self, disperser_client::DisperserClient, BlobStatusRequest, DisperseBlobRequest},
+    errors::EigenDAError,
+};
+
+pub struct EigenDAClient {
+    disperser: Arc<Mutex<DisperserClient<Channel>>>,
+    config: DisperserConfig,
+}
+
+impl EigenDAClient {
+    pub async fn new(config: DisperserConfig) -> Result<Self, EigenDAError> {
+        match rustls::crypto::ring::default_provider().install_default() {
+            Ok(_) => {}
+            Err(_) => {} // This is not an actual error, we expect this function to return an Err(Arc<CryptoProvider>)
```

Comment:
Comment is somewhat unclear, why aren't we making use of this `Arc<CryptoProvider>` maybe something like this is more suitable:
```rs        
        // This might fail if the default provider is already installed
        // So we ignore the result altogether
        let _ = rustls::crypto::ring::default_provider().install_default();
```